xwiki-platform-core/xwiki-platform-wiki/xwiki-platform-wiki-ui/xwiki-platform-wiki-ui-wiki/src/main/resources/WikiManager/WikiUsers.xml

```diff
@@ -519,9 +519,10 @@
     ##
     ## Remove users that have already been invited.
     #set ($alreadyInvited = [])
+    #set ($allInvitations = $services.wiki.user.getAllInvitations($wiki.id))
     #foreach ($name in $names)
-      #foreach ($candidateMember in $services.wiki.user.getAllInvitations($wiki.id))
-        #if ($candidateMember.status == 'PENDING')
+      #foreach ($candidateMember in $allInvitations)
+        #if ("$!name"=="$!candidateMember.getUserId()" &amp;&amp; $candidateMember.status == 'PENDING')
```

Comment:
I think you can write instead:
`#if ($name == $candidateMember.userId &amp;&amp; $candidateMember.status == 'PENDING')`